File: backend/app/schemas/order.py
"""Order schemas — synced with Unit 1."""

from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel, Field


class OrderItemCreate(BaseModel):
    menu_id: int
    quantity: int = Field(..., ge=1)


class OrderCreate(BaseModel):
    items: List[OrderItemCreate] = Field(..., min_length=1)


class OrderItemResponse(BaseModel):
    id: int
    menu_id: int
    menu_name: str
    quantity: int
    unit_price: int
    subtotal: int

    model_config = {"from_attributes": True}


class OrderResponse(BaseModel):
    id: int
    store_code: str
    table_no: int
    session_id: int
    total_amount: int
    status: str
    items: List[OrderItemResponse] = []
    created_at: datetime
    archived_at: Optional[datetime] = None

    model_config = {"from_attributes": True}


class OrderStatusUpdate(BaseModel):
    status: str = Field(..., pattern="^(pending|preparing|completed)$")

File: frontend-admin/src/types/order.ts
export enum OrderStatus { PENDING = 'pending', PREPARING = 'preparing', COMPLETED = 'completed' }
export interface OrderItem { id: number; menuId: number; menuName: string; quantity: number; unitPrice: number; subtotal: number }
export interface Order { id: number; orderNumber: string; storeId: number; tableId: number; sessionId: number; items: OrderItem[]; totalAmount: number; status: OrderStatus; createdAt: string; updatedAt: string }
export interface OrderHistory { id: number; orderNumber: string; tableId: number; tableNumber: number; items: OrderItem[]; totalAmount: number; status: OrderStatus; createdAt: string; archivedAt: string }
export interface OrderStatusCount { pending: number; preparing: number; completed: number }
